crates/fluvio-protocol/src/core/bytebuf.rs

```diff
@@ -0,0 +1,262 @@
+use std::io::Error;
+use std::io::ErrorKind;
+use std::ops::Deref;
+
+use bytes::{Buf, Bytes, BufMut};
+
+use crate::{Encoder, Decoder, Version};
+
+/// Represnts a SmartModule WASM File bytes.
+///
+/// Provides a `Encoder` and `Decoder` implementation optimized for WASM files
+/// used in SmartModules.
+#[derive(Clone, Debug, Default, PartialEq, Eq)]
+pub struct ByteBuf {
+    inner: Bytes,
+}
+
```

Comment:
should have API to create `ByteBuf` from `Bytes`

Reply:
Addressed on: [9d73483](https://github.com/infinyon/fluvio/pull/2738/commits/9d734838757f4ce0408a66bfd9b9a145565b6cc8)